### PR TITLE
[CRUSADE] Upgrade ruby and cleanup ➕

### PR DESCRIPTION
**Pull request by skrix**
### Description
Since there're a lot of abstractions and components, application complexity growth and the only God knows what's going on in production and how everything still works. The main goal of this pull request is to iteratively remove unused parts, increase the readability of code, decrease the number of used abstractions and provide simpler code structure. 

### How to test instructions
Changes will be pushed only to this branch for now and will be updated from time to time. After the point of the ready-to-use version, it will be deployed to the test server. Also, there will be a test suite provided ;)

### Reviews & Collaboration is welcomed and appreciated!

### CODE REVIEW

**Comment by skrix** [comment]
# Crusade Implementation Plan
| Current State   | Plan                                                                                              |
|-----------------|---------------------------------------------------------------------------------------------------|
| app/uploaders   | Move to one uploader service under models/store folder and use shrine as an uploader              |
| app/inputs      | ImagePreviewInput use shrine preview. DateTimeInput use datepicker gem or just customize in place |
| app/helpers     | Reduce using by implementing presenter's mixins or use cells and move under them                  |
| app/policies    | Replace with ez-permissions                                                                       |
| app/services    | Move to logic layer and group by responsibilities with the module as a facade                         |
| app/interactors | Replace with services under logic layer and group by responsibilities                             |
| app/mailer_previews | **_Need more investigation on the usage of this abstraction_**                                          |
| app/forms       | Move under models/forms to reach the incapsulation of persistence logic under models layer        |
| app/finders     | Move under models/queries and provide standard interface                                          |
| app/controllers | Reduce amount of responsibilities by moving logic under logic layer and prepare for migration to graphql |

### If someone have any suggestions or questions feel free to ask me :wink:

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.3.1
+2.6.3
```
Comment (sveredyuk) [inline .ruby-version]: Cool, but the current production server version is 2.3.1 ;)

**Review by sveredyuk — COMMENTED** [review]
Well done. Let's review all this 🎄

**Review by sveredyuk — APPROVED** [review]

**Comment by denys-medynskyi** [comment]
In general, I appreciate 👏 your attitude and plan.

I have also some suggestions:

1. Clean up dead code / unused features first 
**Reasoning**: why we need to refactor code that is not needed? Let's clean up the mess first, and refactor it later

Article for reading: https://kevinjalbert.com/find-and-bury-dead-code/

2. Small PRs instead of 1 gigantic PR
**Reasoning**: it is easier to review, easier to revert, easier to debug issues.

3. Ask for more man-power
**Reasoning**: It is a big amount of work, it will be easier and more fun to do in a bigger group. Also, it can increase the number of active members

- Let's spread the word that we are going to clean up the website in our #genral slack.
- Let's do regular hackathons
- Give away for PRs?

4. Define real-world problems
**Reasoning**: we should focus on making the visitor experience and event admin experience.
I truly agree we need to clean up the code mess, however, we have more problems: *logic* mess.
For example, we have many event/user statuses that are not used.

**Comment by denys-medynskyi** [comment]
https://devcenter.heroku.com/articles/git#deploying-from-a-branch-besides-master

**Comment by denys-medynskyi** [comment]
`git push heroku refactoring:master`

**Comment by denys-medynskyi** [comment]
I think we need to squash existing commits to 1.
It will simplify rebase.

Each refactoring branch should start with this branch.

**Comment by skrix** [comment]
@deny7ko 
> I think we need to squash existing commits to 1.
> It will simplify rebase.

I squashed commits. Yep, it was a good idea. 

> Each refactoring branch should start with this branch.

Of course, next time don't forget this when will start working on a new issue ;) 
But we always have a great rebase 😉

**Comment by skrix** [comment]
The live version of this branch: 
https://pivorak-staging.herokuapp.com/

* we will try to keep it up-to-date but sometimes expectations meet the reality, so ping if notice something outdated

**Pull request merged**